server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -530,18 +530,6 @@ static void validateTypeName(String type) {
         }
         checkIndexSortCompatibility(indexSettings.getIndexSortConfig(), hasNested);
 
-        if (newMapper != null) {
-            DocumentMapper updatedDocumentMapper = newMapper.updateFieldType(fieldTypes.fullNameToFieldType);
-            if (updatedDocumentMapper != newMapper) {
-                // update both mappers and result
-                newMapper = updatedDocumentMapper;
-                results.put(updatedDocumentMapper.type(), updatedDocumentMapper);
-            }
-        }
-
-        // make structures immutable
-        results = Collections.unmodifiableMap(results);
```

Comment:
I'm wondering if this change is important, where we no longer wrap in an unmodifiable map. If not we could restore it to make the PR more targeted/ precise.